src/Juvix/Core/Erased/Evaluator.hs

```diff
@@ -0,0 +1,38 @@
+module Juvix.Core.Erased.Evaluator
+  ( evaluate,
+  )
+where
+
+import Juvix.Core.Erased.Types
+import Juvix.Library hiding (Map, evaluate)
+import Juvix.Utility
+
+evaluate ∷
+  ∀ primVal m.
+  ( HasReader "apply" (primVal → primVal → Maybe primVal) m,
```

Comment:
Nice I'm curious how you got this up, I could't make this from a state monad sadly.

Reply:
Alas, I just haven't made a monad which actually uses this yet.